feat(editor): Improve the status bar item for the VS Code extension by adding a tooltip.

connorshea opened the pull request:
Also do a bit of refactoring.

I found the stats bar item's behavior confusing initially, and I think adding a tooltip will be helpful to understanding what is meant by each state. 

<img width="173" height="89" alt="Screenshot 2025-11-17 at 5 10 47 PM" src="https://github.com/user-attachments/assets/f9b7be37-336b-4311-a1a0-bef40643d0a9" />

<img width="159" height="95" alt="Screenshot 2025-11-17 at 5 11 06 PM" src="https://github.com/user-attachments/assets/fe375a78-b564-4d68-ae23-27bce7b4d2e5" />

<img width="304" height="78" alt="Screenshot 2025-11-17 at 5 11 48 PM" src="https://github.com/user-attachments/assets/d5784d79-2936-41a1-840f-5f47919584a9" />

connorshea commented on editors/vscode/client/extension.ts:
```diff
@@ -310,35 +310,40 @@ export async function deactivate(): Promise<void> {
+    oxcStatusBarItem.show();
+  }
+
+  let [bgColor, icon, tooltipText] = getStatusBarState(enable);
```
If only there were a lint rule for this 🧌

Sysix reviewed (approved):
Looks good!

camc314 commented:
### Merge activity

* **Nov 18, 11:51 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 18, 11:51 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
feat(editor): Improve the status bar item for the VS Code extension by adding a tooltip. (#15819)

Also do a bit of refactoring.

I found the stats bar item's behavior confusing initially, and I think adding a tooltip will be helpful to understanding what is meant by each state.

<img width="173" height="89" alt="Screenshot 2025-11-17 at 5 10 47 PM" src="https://github.com/user-attachments/assets/f9b7be37-336b-4311-a1a0-bef40643d0a9" />

<img width="159" height="95" alt="Screenshot 2025-11-17 at 5 11 06 PM" src="https://github.com/user-attachments/assets/fe375a78-b564-4d68-ae23-27bce7b4d2e5" />

<img width="304" height="78" alt="Screenshot 2025-11-17 at 5 11 48 PM" src="https://github.com/user-attachments/assets/d5784d79-2936-41a1-840f-5f47919584a9" />
